cdc/processor/processor.go

```diff
@@ -106,15 +106,15 @@ func (p *processor) checkReadyForMessages() bool {
 }
 
 // AddTable implements TableExecutor interface.
-func (p *processor) AddTable(ctx cdcContext.Context, tableID model.TableID) (bool, error) {
+func (p *processor) AddTable(ctx cdcContext.Context, tableID model.TableID, startTs model.Ts) (bool, error) {
 	if !p.checkReadyForMessages() {
 		return false, nil
 	}
 
 	log.Info("adding table",
 		zap.Int64("tableID", tableID),
 		cdcContext.ZapFieldChangefeed(ctx))
-	err := p.addTable(ctx, tableID, &model.TableReplicaInfo{})
+	err := p.addTable(ctx, tableID, &model.TableReplicaInfo{StartTs: startTs})
```

Comment:
Could you replace the `if` condition with an assert at the begin of `processor.addTable`?

https://github.com/pingcap/tiflow/blob/3ecf202f6719bbe849dd1d27ef94c41f65d133bc/cdc/processor/processor.go#L923-L929

```diff
 // addTable creates a new table pipeline and adds it to the `p.tables` 
 func (p *processor) addTable(ctx cdcContext.Context, tableID model.TableID, replicaInfo *model.TableReplicaInfo) error { 
 	if replicaInfo.StartTs == 0 { 
- 		replicaInfo.StartTs = p.changefeed.Status.CheckpointTs 
+ 		log.Panic("table start ts must not be 0", zap.Int("tableID", tableID))
 	}

 	if table, ok := p.tables[tableID]; ok { 
```

Reply:
done